app/server.go

```diff
@@ -620,7 +629,7 @@ func NewServer(options ...Option) (*Server, error) {
 	}
 
 	if s.startMetrics {
-		s.SetupMetricsServer()
+		s.platform.RestartMetrics()
```

Comment:
Ah okay, so we are back to this :) Ideally I'd have liked to move this whole thing to the platform service itself but right now it's too difficult.